File: LeetCode/P1Easy_Two_Sum/main.py
from typing import List
# Problem: https://leetcode.com/problems/two-sum/


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        diff = {}

        for i in range(len(nums)):
            item = nums[i]

            if target - item not in diff:
                diff[item] = i
            else:
                return [diff[target - item], i]

File: LeetCode/P1Easy_Two_Sum/main.cpp
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    unordered_map<int, int> seen;

    for (int i = 0; i < nums.size(); i++) {
      if (seen.contains(target - nums[i])) {
        return {i, seen[target - nums[i]]};
      }

      seen[nums[i]] = i;
    }

    return {};
  }
};
